File: solutions_1483485_0/java/poarish/SpeakingInTounges.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package SpeakingInTounges;

import java.io.File;
import java.io.FileWriter;
import java.io.Writer;
import java.util.Scanner;

/**
 *
 * @author Rishabh
 */
public class SpeakingInTounges {
    
    private char[] mapped_array = new char[26];
    
    
    public SpeakingInTounges(){
        for(int i=0;i<mapped_array.length;i++){
            mapped_array[i] = '*';
        }
    }
    
    public void createMapping(String[][] maps){
        int counter = 0;
        for(int i=0;i<maps.length;i++){
            String goog = maps[i][0];
            String normal = maps[i][1];
            for(int j=0;j<goog.length();j++){
                int index = goog.charAt(j) - 'a';
                if(goog.charAt(j) != ' ' && mapped_array[index] == '*'){
                    counter++;
                    mapped_array[index] = normal.charAt(j);
                }
                else if(goog.charAt(j) != ' ' && normal.charAt(j) != mapped_array[index]){
                    System.out.println("Its an error");
                    System.exit(0);
                }
            }
        }
//        System.out.println("Done for " + counter + " characters");
//        for(int i=0;i<mapped_array.length;i++){
//            System.out.println((char)('a'+i) + " is mapped to " + mapped_array[i]);
//        }
    }
    
    public void readFile(String path) throws Exception{
        File f = new File(path);
        Scanner scanner = new Scanner(f);
        String name = f.getName().substring(0,f.getName().indexOf('.'));
        Writer writer = new FileWriter(name+".out");
        int n = Integer.parseInt(scanner.nextLine());
        int count=0;
        while(scanner.hasNextLine()){
            count++;
            writer.write("Case #"+count+": "+convert(scanner.nextLine())+"\r\n");
        }
        writer.close();
        scanner.close();
    }
    
    private String convert(String s){
        String ans = "";
        for(int i=0;i<s.length();i++){
            if(s.charAt(i) == ' '){
                ans += " ";
            }
            else{
//                System.out.println(s.charAt(i));
                ans += String.valueOf(mapped_array[s.charAt(i)-'a']);
            }
        }
        return ans;
    }
    
    public static void main(String args[]) throws Exception{
        String[][] maps = new String[][]{ 
            new String[]{
                "ejp mysljylc kd kxveddknmc re jsicpdrysi",
                "our language is impossible to understand"
            }, 
            new String[]{
                "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd",
                "there are twenty six factorial possibilities"
            }, 
            new String[]{
                "de kr kd eoya kw aej tysr re ujdr lkgc jv",
                "so it is okay if you want to just give up"
            },
            new String[]{
                "zq",
                "qz"
            }
        };
        SpeakingInTounges speakingInTounges = new SpeakingInTounges();
        speakingInTounges.createMapping(maps);
        speakingInTounges.readFile("A-small-practice.in");
    }
    
}

File: solutions_1483485_0/C#/Seph/SpeakingInTounges.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;

namespace GoogleCodeJam
{
    class SpeakingInTounges
    {

        public static void Main()
        {
            string InputPath = "C:\\Users\\Seph\\Documents\\Visual Studio 2010\\Projects\\GoogleCodeJam\\GoogleCodeJam\\SpeakingInTounges\\in.txt";
            string OutputPath = "C:\\Users\\Seph\\Documents\\Visual Studio 2010\\Projects\\GoogleCodeJam\\GoogleCodeJam\\SpeakingInTounges\\out.txt";

            char[] mapping = new char[26];

            string input = "ejp mysljylc kd kxveddknmc re jsicpdrysi rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd de kr kd eoya kw aej tysr re ujdr lkgc jv";
            string output = "our language is impossible to understand there are twenty six factorial possibilities so it is okay if you want to just give up";

            for (int x = 0; x < input.Length; x++)
            {
                char cIn = input[x];
                char cOut = output[x];
                if (cIn != ' ')
                    mapping[cIn - 'a'] = cOut;
            }
            mapping['y' - 'a'] = 'a';
            mapping['e' - 'a'] = 'o';
            mapping['q' - 'a'] = 'z';
            mapping['z' - 'a'] = 'q';
            //Console.WriteLine(mapping);

            string FileOutput = "";
            string FileInput = System.IO.File.ReadAllText(InputPath);

            string[] InputLines = FileInput.Split(new string[] { "\r", "\n", "\r\n" }, StringSplitOptions.RemoveEmptyEntries);

            int Cases = Int32.Parse(InputLines[0]);
            for (int x = 1; x <= Cases; x++)
            {
                if (x != 1)
                    FileOutput += "\n";
                FileOutput += "Case #"+x+": "+Translate(mapping,InputLines[x]);
            }


            System.IO.File.WriteAllText(OutputPath, FileOutput);

        }

        public static string Translate(char[] mapping, string input)
        {
            string result = "";

            for (int x = 0; x < input.Length; x++)
            {
                char cur = input[x];
                char replace = cur!=' '?mapping[cur - 'a']: cur;
                result += replace;
            }

            return result;

        }
    }
}
